src/gdb_main.c

```diff
@@ -264,10 +264,15 @@ int gdb_main_loop(target_controller_s *tc, char *pbuf, size_t pbuf_size, size_t
 			/* Skip past the '=' and convert the value */
 			++packet;
 			const size_t value_length = strlen(packet) / 2U;
+			/* If the value is bigger than 4 bytes report error */
+			if (value_length > 4) {
```

Comment:
This and the above should ideally use `U` suffixed constants so the comparison is always done unsigned.

Looks good otherwise with the note that we should make the value compared here depend on the target so it works with 64-bit targets too, or upgrade the buffer to 8 bytes always, but this is for later work and not a problem for this PR (unless you want to tackle it!).

Reply:
I will fix the constant. I think the target dependence stuff should be dealt with separately.

Reply:
Aye, we mostly wrote the second half of that so it's noted somewhere 😄